types/react-draft-wysiwyg/index.d.ts

```diff
@@ -8,18 +8,217 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.9
 
-import * as React from 'react';
 import * as Draft from 'draft-js';
+import * as React from 'react';
 
 export type SyntheticKeyboardEvent = React.KeyboardEvent<{}>;
 export type SyntheticEvent = React.SyntheticEvent<{}>;
 export type RawDraftContentState = Draft.RawDraftContentState;
+export type DraftStyleMap = Draft.DraftStyleMap;
 
 export class EditorState extends Draft.EditorState {}
 export class ContentState extends Draft.ContentState {}
 export class ContentBlock extends Draft.ContentBlock {}
 export class SelectionState extends Draft.SelectionState {}
 
+export interface ToolbarCommonProps {
+    icon?: string;
+    className?: string;
+    title?: string;
+}
+
+export interface MentionsSuggestions {
+    text?: string;
+    value?: string;
+    url?: string;
+}
+
+export interface EditorMentions {
+    separator?: string;
+    trigger?: string;
+    suggestions?: MentionsSuggestions[];
+}
+
+export interface EditorHashtag {
+    separator?: string;
+    trigger?: string;
+}
+
+export type ToolbarOptions =
+    | 'inline'
+    | 'blockType'
+    | 'fontSize'
+    | 'fontFamily'
+    | 'list'
+    | 'textAlign'
+    | 'colorPicker'
+    | 'link'
+    | 'link'
+    | 'embedded'
+    | 'emoji'
+    | 'image'
+    | 'remove'
+    | 'history';
+
+export type ToolbarInlineOptions =
+    | 'bold'
+    | 'italic'
+    | 'underline'
+    | 'strikethrough'
+    | 'monospace'
+    | 'superscript'
+    | 'subscript';
+
+export type ToolbarBlockTypeOptions = 'Normal' | 'H1' | 'H2' | 'H3' | 'H4' | 'H5' | 'H6' | 'Blockquote' | 'Code';
+export type ToolbarListOptions = 'unordered' | 'ordered' | 'indent' | 'outdent';
+export type ToolbarTextAlignOptions = 'left' | 'center' | 'right' | 'justify';
+export type ToolbarLinkOptions = 'link' | 'unlink';
+export type ToolbarHistoryOptions = 'undo' | 'redo';
+export type ToolbarComponent = React.ComponentType<any>;
+
+export interface EditorToolbar {
+    options?: ToolbarOptions[];
+    inline?: {
+        inDropdown?: boolean;
+        className?: string;
+        component?: ToolbarComponent;
```

Comment:
For the self defined component，should we allow self defined properties？For this purpose you can define the `inline` type as a generic type, like `interface InlineProps<T> extends T {}`?